python/ray/serialization.py

```diff
@@ -476,6 +480,9 @@ def serialize(self, value):
             return RawSerializedObject(value)
 
         if self.worker.use_pickle:
```

Comment:
Can't just check version?

Reply:
Checking the version is hard: https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python and I don't want to depend on another module.

Reply:
Tested this and didn't work for 1.15.4, so changed to something that does.

Reply:
Ah, not sure if I like that, if we test for a feature we should test for the one that is relevant. The problem here is that I tested on the np.ndarray type, but it needs to be tested on an object of that type. It's awkward to instantiate one, but maybe we do it in the constructor?

The version check is also somewhat slow unfortunately, so we should avoid doing that:
```
In [1]: from packaging import version                                                                                                                                                                                                                         

In [2]: %timeit version.parse("2.3.1") < version.parse("10.1.2")                                                                                                                                                                                              
13.9 µs ± 341 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [3]: %timeit hasattr(version, "parse")                                                                                                                                                                                                                     
77.1 ns ± 5.39 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
Maybe it's ok.

Reply:
Any reason not to just parse `numpy.__version__`? Would break if numpy changes their version string format but that seems like a very minor concern.

Reply:
But anyways performance shouldn't be a concern - we should probably move the check into the worker instead of in this critical section anyways.

Reply:
Ok yeah, I'll move it into the worker and use the `packaging` package just to be save (we should avoid parsing the version ourselves).

Reply:
Sounds great, please feel free to merge and ping me once tested

Reply:
Oh also be sure that it works for both `ray.init()` and `ray start` (I think adding to `connect()` should do the trick?)